### PR TITLE
Make case classes Serializable

### PR DESCRIPTION
**Pull request by OlivierBlanvillain**

### CODE REVIEW

File: tests/run/case-class-serializable.scala [hunk 1]
```diff
@@ -0,0 +1,6 @@
+case class Foo()
+
+object Test {
+  def main(args: Array[String]): Unit =
+    assert(Foo().isInstanceOf[Serializable])
```
Comment (smarter) [inline tests/run/case-class-serializable.scala]: I would go further and make the test actually serialize and deserialize a case class and verify the instances are ==

Comment (OlivierBlanvillain) [reply tests/run/case-class-serializable.scala]: Done, we can probably reuse parts of that test for lambdas...

**Comment by OlivierBlanvillain** [comment]
@nicolasstucki How do I regenerate these files:?

- run/literals.decompiled
- pos/i2104b.decompiled
- pos/simpleCaseObject.decompiled
- pos/simpleCaseClass-3.decompiled
- pos/simpleCaseClass-2.decompiled
- pos/simpleCaseClass-1.decompiled

**Comment by nicolasstucki** [comment]
Run `testFromTasty i2104b`, it will generate `pos/i2104b.decompiled.out`.

**Comment by smarter** [comment]
We should really automatize this: in scalac, `partest --update-check` will automatically update all failing check files.

**Comment by OlivierBlanvillain** [comment]
```sh
$ sbt "testFromTasty i2104b" && mv pos/i2104b.decompiled{.out,}
...
mv: cannot stat 'pos/i2104b.decompiled.out': No such file or directory
```

**Comment by smarter** [comment]
tests/pos instead of pos ?

**Comment by liufengyun** [comment]
Should we avoid adding `Serializable` if the case class already extends `Serializable`?

**Comment by smarter** [comment]
@liufengyun Yes, just like scalac we should allow explicitly extending Serializable or Product in a case class without error, can you create an issue for that ?

**Pull request merged**